### DISCUSSION TITLE
Open with external IP instead localhost

### DISCUSSION THREAD

**Discussion by De-M-oN** [rejetto/hfs#149] [General]
Can I somehow make HFS open the user interface with external IP and not localhost?
Because this way I always have to edit the url of the frontend to share files.

**Comment by rejetto** [comment]
you mean when you start HFS and it opens the Admin panel ?

**Comment by De-M-oN** [comment]
Yes. It opens with localhost as url then.
In HFS 2.3 I could choose that it uses my DNS or an other IP

In HFS 2.3 I could this way just mark in admin panel all files I want to share and press CTRL+C and could paste the URLs (works still via workaround in user interface by making a zip out of it, but still. Was very convenient and this is missing now too. Drag&Drop for adding files is missing too. If it can somehow later return would be cool.

But with the localhost being the default IP is the more "annoying" one :D

**Comment by rejetto** [comment]
you can choose your dns/ip in hfs3 too.
go to "shared files" and click on any file.
you can see the "link" field, with the "copy" button.
if you don't like the address, you just use the other button to change it
<img width="575" alt="image" src="https://user-images.githubusercontent.com/1367199/224558588-58388593-9de3-4762-9bdd-462d586500ae.png">
maybe you didn't notice this feature because it was introduced lately.


drag&drop to add: I don't know if it's possible, because the browser doesn't let me see the real path of the file you drag. So, it is possible that we may never see this. If you find how to do it let me know.
But maybe in the future, to overcome this problem, we may add a "right click on file" feature. That should be similarly convenient, don't you think?

**Comment by De-M-oN** [comment]
Ok so copying files in the admin panel is now fine
Frontend for the user still is a localhost link in the admin panel. Not the end of the world, just have to remember myself to copy files in the admin panel then. me acquired bad habits by not knowing about the base url in the file properties. so thank you for that 😁 

Ok I thought it is possible because e.g. Youtube or some file host sites etc allow to drag&drop files. But I'm no programmer so I cant help here.

**Comment by rejetto** [comment]
HFS too supports drag&drop if you want to upload.
When you upload you just need the content of the file, you don't need to know the full path where it is stored on your hard disk.

Adding to the VFS is a different thing, tho. I need the path of your hard disk, and the browser won't tell me that.

**Reply by De-M-oN** [reply]
Ah ok makes sense

**Comment by rejetto** [comment]
> Frontend for the user still is a localhost link in the admin panel. 

I guess here you meant the home page of the admin panel, where the link is actually localhost.

> Not the end of the world, just have to remember myself to copy files in the admin panel then.

i guess here by "admin panel" you meant the "shared files" page. Right?

**Reply by De-M-oN** [reply]
both yes.

**Comment by rejetto** [comment]
anyway, I'm interested in the possibility to make things easier or clearer, in general, so if you have suggestions let me know.
